replace hard locking to Guzzle Stream with PSR StreamInterface

jasverix opened the pull request:
Guzzle::streamFor() already accepts StreamInterface and does not require Guzzle Stream.

We are using another provider (nyholm) for streams in our project and the requirement for Guzzle Stream is really a blocking issue for us. And we cannot see any reason for that, because Guzzle itself only requires StreamInterface.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/msgraph-sdk-php/pull/581)

Ndiritu commented:
First, thanks for taking time to improve this library @jasverix! 🎉 

A pointer:
Our models are [auto-generated](https://github.com/microsoftgraph/msgraph-sdk-php/blob/dev/CONTRIBUTING.md#how-the-library-is-built) therefore these changes will be made in that repo.

However, we can still discuss the issue you're facing here...

The getters currently return a Psr StreamInterface impl (albeit Guzzle Stream) in all cases I've thought of i.e. GuzzleHttp\Psr7\Utils::streamFor() returns a Guzzle Stream which is Psr compliant

Please help me understand how using the Nyholm stream implementation doesn't work with this library?

I would however agree with changing our PHPDocs to StreamInterface (both setters and getters) to make things clearer and agree with the if-statement type check changing to StreamInterface

jasverix commented:
Hi,

I might have overreacted here. My problem is the attachBody method in the GraphRequest class. I can update this PR and keep only that change.

jasverix commented:
I have now fixed the branch. I just did a big replace operation, but my problem is the `attachBody` method. We are passing a file to OneDrive, and we already have that file as a StreamInterface, but not a Guzzle Stream. Guzzle handles it perfectly, but attachBody had a `is_a()` check that I've now changed.

Ndiritu reviewed (changes requested):
Looks good.

jasverix commented on composer.json:
```diff
@@ -14,7 +14,8 @@
 		"guzzlehttp/guzzle": "^6.0 || ^7.0",
-		"ext-json": "*"
+		"ext-json": "*",
+		"psr/http-client": "^1.0"
```
Changed to correct package now.

How can we let guzzle determine the version? Right now, we have the exact same version query as guzzle (^1.0).

jasverix replied:
I can use a string, as you did before, instead of the `::class` constant, and remove the package requirement, if you want.

I just, in general, personally like code better than strings, but it will do no big difference in this case.

Ndiritu replied:
> I can use a string, as you did before, instead of the `::class` constant, and remove the package requirement, if you want.
> 
> I just, in general, personally like code better than strings, but it will do no big difference in this case.

I also prefer the `::class` constant. We'll be removing the string in v2.
